### ISSUE TITLE
Passing any arguments to authorization calls causes an error

### ISSUE THREAD

**Issue by fsavina** [spatie/laravel-permission#720]
I've recently updated to the 2.12.0 release without any change to my code.
I've a model policy and I authorize actions in my controller:

```php
public function update(Request $request, Post $post)
{
    $this->authorize('update', $post);
}
```

The package is now causing an error in the Permission registrar, when registering the `before` gate hook.
It is trying to use the model ($post) as a guard name:

```php
public function registerPermissions(): bool
{
    $this->gate->before(function (Authorizable $user, string $ability, $guard = null) {
        if (\is_array($guard)) {
            $guard = array_first($guard);
        }
        
        try {
            if (method_exists($user, 'hasPermissionTo')) {
                return $user->hasPermissionTo($ability, $guard) ?: null;
            }
        } catch (PermissionDoesNotExist $e) {
        }
    });
    
    return true;
}
```

This means that I'm forced to change any authorization call by passing the guard name before any other parameter. This shouldn't be a backward compatible update.

**Comment by drbyte** [comment]
> This means that I'm forced to change any authorization call by passing the guard name before any other parameter. This shouldn't be a backward compatible update.

Argh. 
You're right.
The testing focused mostly around calls to the gate via `can()`, not `authorize` as used with a policy.

Stick with v2.11 until we sort this out.  Apologies!

**Comment by fsavina** [comment]
No worries, thanks!

**Comment by drbyte** [comment]
@fsavina I have a short bit of time to look into this more deeply in about half an hour. But I don't have a project using `authorize()` calls that way handy to test with (although I realize it's not terribly complicated). Are you able to share a copy of your code base (even privately, click my name above to get my email address from my GH profile); that'll help ensure I catch any edge cases too ;)

**Comment by fsavina** [comment]
I'm afraid I can't because it's an internal project under development and I haven't been told yet if it's gonna be release as open source or not.

Anyways the problem is caused by a [resource controller](https://laravel.com/docs/5.6/controllers#resource-controllers) and [model policy](https://laravel.com/docs/5.6/authorization#creating-policies) developed according to the standard docs.

Any action in the controller receives a model instance on which we call the `authorize` method as shown above and as indicated [here](https://laravel.com/docs/5.6/authorization#authorizing-actions-using-policies).

I'm quite confident that you should be able to reproduce the error simply by following the examples in the Laravel auth docs page. Let me know if you need more info.

**Comment by drbyte** [comment]
Ya, I was thinking the exact same. Then remembered that I have policies set up in my permissions-demo repo ... and yet all its tests are passing when using 2.12.0

https://github.com/drbyte/spatie-permissions-demo/blob/master/app/Policies/PostPolicy.php
https://github.com/drbyte/spatie-permissions-demo/blob/master/app/Providers/AuthServiceProvider.php#L19

Maybe it's because it's using `authorizeResource()`, instead of writing an authorize() call in each method: https://github.com/drbyte/spatie-permissions-demo/blob/master/app/Http/Controllers/PostsController.php#L13

**Comment by drbyte** [comment]
Nope. authorizeResource() passes, and so does using individual authorize() calls. 

Granted, this demo app is only using the default `web` guard; it doesn't demo any specialized guard functionality of the spatie package.

So, not sure what's different between this demo app and yours.

**Comment by daguilarm** [comment]
Same error... my Controller:

Gate::denies('create plot', Plot $plot);

I'm back to 2.11.0 and no problems

**Comment by drbyte** [comment]
Released 2.12.1 which reverts the 2.12.0 changes. Apologies!
